### ISSUE TITLE
Repeat is not working for some commands

### ISSUE THREAD

**Issue by weirdNox** [cyrus-and/gdb-dashboard#128]
Hello there! Great package you have here :)

Recently, after updating GDB to version 8.2, I am unable to repeat commands like `next`, `step`, `stepi`...

Also, this only seems to be affected when the program is running. When no program is running, executing `next` and then pressing enter without any input, will repeat the command and show the error that no program is running. However, when the program is running, it will only insert a new `>>>` prompt, with no error or redisplay.

Other commands like `break` repeat, so I'm thinking that the redisplay is doing something that makes GDB not repeat the command.

Without gdb-dashboard, the repeat works fine, so it must be something that changed with the update.

I'll try to rollback the update and see if it works.

**Comment by weirdNox** [comment]
Yes, I just downgraded GDB to 8.1.1 and it worked fine, so the update must have changed the python API in some way

**Comment by cyrus-and** [comment]
> Great package you have here :)

Hey, thanks!

Yes, I can reproduce that, that's weird, I suspect that GDB commands executed from Python somehow invalidate the last command, thus spoiling the repeat feature. Let me look into this...

**Comment by cyrus-and** [comment]
Yes that seems to be the case, try running `python gdb.execute('echo asd')` then press enter to repeat.

In GDB 8.2:

```
(gdb) python gdb.execute('echo asd')
asd(gdb)
(gdb)
(gdb)
(gdb)
```

Prior versions:

```
(gdb) python gdb.execute('echo asd')
asd(gdb)
asd(gdb)
asd(gdb)
asd(gdb)
```

The [`gdb.execute`][1] function has a `from_tty` flag:

> The from_tty flag specifies whether GDB ought to consider this command as having originated from the user invoking it interactively. It must be a boolean value. If omitted, it defaults to False.

In the dashboard it's used like this:

https://github.com/cyrus-and/gdb-dashboard/blob/62036259ed6423917354781a57e97629a40a1457/.gdbinit#L129-L130

So `from_tty` is `False` and it should be enough to tell GDB to not let this command invocation to hijack the last proper TTY command. I suspect this is a regression bug, feel free to report it to the GDB team and post the link here.

[1]: https://sourceware.org/gdb/onlinedocs/gdb/Basic-Python.html#index-gdb_002eexecute-1

**Comment by weirdNox** [comment]
Yes, that seems to be it, great catch! 

In this case, disabling both the disassembly and registers views (the ones which use the `run` function), the repeat feature is restored.

I'll see how to report a bug in GDB, and then report back :)

**Comment by cyrus-and** [comment]
Did you manage to file that issue?

**Comment by cirosantilli** [comment]
I wonder if it would be possible to implement a workaround for this in the gdb-dashboard code, this thing is killing me on Ubuntu 18.10 :-)

**Comment by cyrus-and** [comment]
@cirosantilli eh, I understand... I'am afraid this cannot be fixed Python-side (but feel free to share ideas about it).

Anyway, the upstream issue appears to be fixed now so worst case you can just recompile your GDB.

**Comment by nicolasmarie1** [comment]
For those who have this issue in 2024 because there are stuck on Red Hat 8, it is possible to circumvent the issue by redefining the command like so:

```
define n
    next
end
```
